pkg/config/teeconfig/teeconfig.go

```diff
@@ -111,91 +124,209 @@ func (t *teeConfig) ParseEnvAsSlice(key string, fn func(string) []interface{}) {
 
 // IsSet wraps Viper for concurrent access
 func (t *teeConfig) IsSet(key string) bool {
-	return t.baseline.IsSet(key)
+	base := t.baseline.IsSet(key)
+	compare := t.compare.IsSet(key)
+	if base != compare {
+		log.Warnf("difference in config: IsSet(%s) -> base: %v | compare %v", key, base, compare)
+	}
+	return base
 }
 
 func (t *teeConfig) AllKeysLowercased() []string {
-	return t.baseline.AllKeysLowercased()
+	base := t.baseline.AllKeysLowercased()
+	compare := t.compare.AllKeysLowercased()
+	slices.Sort(base)
+	slices.Sort(compare)
+	if !reflect.DeepEqual(base, compare) {
+		log.Warnf("difference in config: allkeyslowercased() -> base len: %d | compare len: %s", len(base), len(compare))
+		d := 0
+		for i := range base {
+			if i < len(compare) {
+				if base[i] == compare[d] {
+					d++
+				} else {
+					log.Warnf("AllKeysLowercased item %d *** : %s (%v) | %s", i, base[i], t.baseline.Get(base[i]), compare[d])
+				}
+			}
+			log.Flush()
+		}
+
+	}
+	return base
 }
 
 // Get wraps Viper for concurrent access
 func (t *teeConfig) Get(key string) interface{} {
-	return t.baseline.Get(key)
+	base := t.baseline.Get(key)
+	compare := t.compare.Get(key)
+	if !reflect.DeepEqual(base, compare) {
```

Comment:
This block of code is generic enough that we could make it into a helper:

```
func compareResult(key, method string, base, compare interface{}) interface {}
```

It may be worth using `runtime.Caller` to get the call-site as well, which would help in understanding usage patterns.

We don't need to worry about scrubbing, do we? This should only run if its enabled in staging, which should be safe.

Reply:
logs are scrubbed, so we're fine.